src/lib.rs

```diff
@@ -73,6 +75,37 @@ impl ReplicaId {
 #[derive(PartialEq, Debug, Clone, Default)]
 pub struct Layer(pub Vec<Node>);
 
+impl From<&Vec<u8>> for Layer {
+    fn from(data: &Vec<u8>) -> Self {
+        let mut nodes = Vec::with_capacity(data.len() / NODE_SIZE);
+        let mut temp = [0u8; NODE_SIZE];
+        for slice in data.chunks_exact(NODE_SIZE) {
+            temp.copy_from_slice(&slice[..]);
+            nodes.push(Node(
+                Fr::from_repr(unsafe { std::mem::transmute::<[u8; NODE_SIZE], FrRepr>(temp) })
+                    .unwrap(),
+            ));
+        }
+        Layer(nodes)
+    }
+}
+
+impl From<&Layer> for Vec<u8> {
+    fn from(layer: &Layer) -> Self {
+        let mut ret = Vec::with_capacity(layer.0.len() * NODE_SIZE);
+        for n in layer.0.iter() {
+            ret.extend(unsafe {
+                std::slice::from_raw_parts(
+                    &n.0.into_repr() as *const FrRepr as *const u8,
+                    std::mem::size_of::<FrRepr>(),
+                )
+                .to_vec()
+            });
```

Comment:
Same as above, there needs to be a check that `NODE_SIZE` and `std::mem::size_of::<FrRepr>` are the same.